sdk/core/azure-core/src/main/java/com/azure/core/http/ProxyOptions.java

```diff
@@ -23,9 +25,40 @@
  * This represents proxy configuration to be used in http clients..
  */
 public class ProxyOptions {
-    private static final ClientLogger LOGGER = new ClientLogger(ProxyOptions.class);
+    /**
+     * Represents a list of hosts that should be reached directly, bypassing the proxy.
+     * This is a list of patterns separated by '|'. The patterns may start or end with a '*' for wildcards.
+     * Any host matching one of these patterns will be reached through a direct connection instead of through a proxy.
+     *
+     * Default value is {@code null}
+     */
+    public static final String HTTP_PROXY_NON_PROXY_HOSTS = "http.proxy.non-proxy-hosts";
+
+    /**
+     * The HTTP host name of the proxy server.
+     * Default value is {@code null}.
+     */
+    public static final String HTTP_PROXY_HOST = "http.proxy.hostname";
+
+    /**
+     * The port number of the proxy server.
+     * Default value is 443.
+     */
+    public static final String HTTP_PROXY_PORT = "http.proxy.port";
+
+    /**
+     * The HTTP proxy server user.
+     * Default value is {@code null}.
+     */
+    public static final String HTTP_PROXY_USER = "http.proxy.username";
 
-    private static final String INVALID_CONFIGURATION_MESSAGE = "'configuration' cannot be 'Configuration.NONE'.";
+    /**
+     * The HTTP proxy server password.
+     * Default value is {@code null}.
+     */
+    public static final String HTTP_PROXY_PASSWORD = "http.proxy.password";
```

Comment:
Do we require username also to be set when password is set similar to port and hostname (in AMQP proxy options)?

Reply:
I want to keep non-code configuration consistent with in-code configuration: null and correctness validation should be taken care of regardless of how the value is passed. In our case it happens here for both cases.

https://github.com/Azure/azure-sdk-for-java/blob/a1bf04a61df60ab8d9cfc84c9ff86b24ad5ffec7/sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/ProxyOptions.java#L139-L144

AMQP port is special because there is no default value and it needs to be converted to primitive `int` - there is no validation needed for in-code configuration case.